gsa/src/web/graphql/policies.js

```diff
@@ -165,3 +181,56 @@ export const useClonePolicy = options => {
   const policyId = data?.clonePolicy?.id;
   return [clonePolicy, {...other, id: policyId}];
 };
+
+export const useCreatePolicy = options => {
+  const [queryCreatePolicy, {data, ...other}] = useMutation(
+    CREATE_POLICY,
+    options,
+  );
+  const createPolicy = useCallback(
+    // eslint-disable-next-line no-shadow
+    (inputObject, options) =>
+      queryCreatePolicy({...options, variables: {input: inputObject}}).then(
+        result => result?.data?.createPolicy?.id,
+      ),
+    [queryCreatePolicy],
+  );
+  const policyId = data?.createPolicy?.id;
+  return [createPolicy, {...other, id: policyId}];
+};
+
+export const useLoadPolicyPromise = () => {
+  const client = useApolloClient();
+
+  const loadPolicy = policyId =>
+    client
+      .query({
+        query: GET_POLICY,
+        variables: {id: policyId},
+        fetchPolicy: 'no-cache', // do not cache, since this is used when a change is saved
+      })
+      .then(response => {
+        const policy = Policy.fromObject(response?.data?.policy);
+
+        return policy;
+      });
+
+  return loadPolicy;
+};
```

Comment:
```suggestion
export const useLazyGetPolicy = () => {
  const client = useApolloClient();
  let policy;
  const getPolicy = policyId =>
    client
      .query({
        query: GET_POLICY,
        variables: {id: policyId},
        fetchPolicy: 'no-cache', // do not cache, since this is used when a change is saved
      })
      .then(response => {
        if (isDefined(response?.data?.policy) {
          policy = Policy.fromObject(response?.data?.policy);
        }

        return policy;
      });

  return [getPolicy, policy];
};
```

Reply:
I don't think this works because queries are not promises and we cannot use the .then(() => ...)